xarray/testing/assertions.py

```diff
@@ -95,6 +95,18 @@ def assert_isomorphic(a: DataTree, b: DataTree, from_root: bool = False):
         raise TypeError(f"{type(a)} not of type DataTree")
 
 
+def align_dims(a, b, check_dims):
+    if check_dims not in ("strict", "transpose"):
+        raise ValueError(f"Invalid value for check_dims: {check_dims}")
+    if not isinstance(a, (Variable, DataArray, Dataset)):
+        return b
+    if check_dims == "transpose":
+        assert set(a.dims) == set(b.dims), f"Dimensions differ: {a.dims} {b.dims}"
+        return b.transpose(*a.dims)
+    assert a.dims == b.dims, f"Dimensions differ: {a.dims} {b.dims}"
```

Comment:
This is good - if the dimensions differ there is no point checking anything else.

Reply:
Are you sure we don't have better diffing functions further down the stack? We should ensure the output of an error is no worse with this than with this commented out

Reply:
The usual diff would allow you to see if two datasets somehow got created with the same values but swapped dimensions. If we want this, I can skip the transpose when check_dim_order=False and the dims differ (so it doesn't fail on `transpose`) and just let it make the whole diff. And I'll always skip the dims check when check_dim_order=True.

Reply:
Yes I would probably let it fall through, so we have a single place where we check & generate diffs; at least unless there's a good reason not to

(though @TomNicholas let us know if you disagree!)